marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/matcher-impl/matcher-impl.xqy

```diff
@@ -427,10 +427,12 @@ declare function match-impl:values-by-qname(
 {
   map:new(
     for $qname in fn:distinct-values((map:get($compiled-options, "queries") ! map:get(.,"qname")))
-    let $values := fn:distinct-values($document//*[fn:node-name(.) eq $qname] ! fn:normalize-space(fn:string(.))[.])
+    let $values := fn:distinct-values(
+      $document//*[fn:node-name(.) eq $qname][fn:normalize-space(fn:string(.))]/fn:data(.)
```

Comment:
Is the addition of "fn:data" the key change here? 

@ryanjdew  I'm good with this if you are.

Reply:
Yes. We now filter on blank values, but ignore the string cast and get the raw data if it passes the test.

There is a possible change in behavior here: a string-valued node with extra whitespace would have been trimmed, previously (_e.g._ `  this   has spaces ` ⇒ `this has spaces`), and now won’t be.

Reply:
Note that only the value to search for is changed—the targets are not changed in any way. Under the old behavior, a document might not even match itself. I would argue this change is for the better; data sanitization should be intentional and explicit, not automagical.

Reply:
I'm good with this change.